tests/test_server_basic.py

```diff
@@ -718,6 +718,95 @@ def test_list_file(dispatcher_live_fixture):
     assert job_id == calculated_job_id
 
 
+def test_catalog_file(dispatcher_live_fixture):
+    server = dispatcher_live_fixture
+    logger.info("constructed server: %s", server)
+
+    # let's generate a valid token
+    token_payload = {
+        **default_token_payload,
+        "roles": "unige-hpc-full, general",
+    }
+    encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    params = {
+        **default_params,
+        'product_type': 'dummy',
+        'query_type': "Dummy",
+        'instrument': 'empty',
+        'token': encoded_token
+    }
+
+    file_path = DispatcherJobState.create_catalog_file(catalog_value=5)
+
+    list_file = open(file_path)
+
+    jdata = ask(server,
+                params,
+                expected_query_status=["done"],
+                max_time_s=150,
+                method='post',
+                files={"user_catalog_file": list_file.read()}
+                )
+
+    list_file.close()
+    assert 'user_catalog_file' in jdata['products']['analysis_parameters']
+    # test job_id
+    job_id = jdata['products']['job_id']
+    session_id = jdata['session_id']
+    # adapting some values to string
+    for k, v in params.items():
+        params[k] = str(v)
+
+    restricted_par_dic = InstrumentQueryBackEnd.restricted_par_dic({**params, "user_catalog_file": f'temp_sid_{session_id}/user_catalog_file', "sub": "mtm@mtmco.net"})
+    calculated_job_id = make_hash(restricted_par_dic)
+
+    assert job_id == calculated_job_id
+
+
+def test_user_catalog(dispatcher_live_fixture):
+    server = dispatcher_live_fixture
+    logger.info("constructed server: %s", server)
+
+    # let's generate a valid token
+    token_payload = {
+        **default_token_payload,
+        "roles": "unige-hpc-full, general",
+    }
+    encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    selected_catalog = """{"cat_column_descr":[["meta_ID","<i8"],["src_names","<U20"],["significance",">f4"],["ra",">f4"],["dec",">f4"],["NEW_SOURCE",">i2"],["ISGRI_FLAG","<i8"],["FLAG","<i8"],["ERR_RAD","<f8"]],"cat_column_list":[[1],["Test A"],[6.233789443969727],[299.81768798828125],[40.72419357299805],[-32768],[2],[0],[0.000029999999242136255]],"cat_column_names":["meta_ID","src_names","significance","ra","dec","NEW_SOURCE","ISGRI_FLAG","FLAG","ERR_RAD"],"cat_coord_units":"deg","cat_frame":"fk5","cat_lat_name":"dec","cat_lon_name":"ra"}"""
```

Comment:
could this line be shortened?

Reply:
is it certain that's what oda_api sends? I do not remember. There should be a test with oda_api somewhere too.

Reply:
I will provide a test also with oda_api

Reply:
> could this line be shortened?

Perhaps, but I don't know much regarding the way catalogs are formatted. Is there a place I can look into to understand more?

Reply:
> > could this line be shortened?
> 
> Perhaps, but I don't know much regarding the way catalogs are formatted. Is there a place I can look into to understand more?

you can look into how oda_api sends it.